### ISSUE TITLE
Interpolation layer skips padding

### ISSUE THREAD

**Issue by ClePol** [Deep-MI/FastSurfer#343]
## Description
In the FastSurferVINN interpolation layer, images are cropped in the first interpolation layer, but not padded in the second interpolation layer. This causes the VINN network to crash, since the size of skip connections with full resolution does not match the non-padded layer output.

## Steps to Reproduce
Create a dataset where all slices has the same resolution and size (causing scale factors to be always 1) and run training.
I am not sure why this does not happen during normal inference or training. Code to reproduce is available, unfortunately in my customized FastSurfer version.

## Expected Behavior
No crash, FastSurferVINN runs as usual.

## Environment
 - FastSurfer Version: Customized FastSurfer V2.1.0, vanilla FastSurferVINN network architecture
 - OS: Ubuntu 20.04.5 LTS

### Execution
Local validation:
python3 run_model.py --cfg config/debug.yaml

## Additional Context
Already looked at the code with @dkuegler to confirm this is a bug and not an error in local changes.

**Comment by dkuegler** [comment]
Confirmed
-----------

I think just scale factor always 1 is not sufficient to cause this condition, additionally, input images have to be padded through the `MultiResolutionDataset`. That is probably why this has not happened before, either at least one scale factor was not 1 (usually in training) or the images are not padded (usually at inference time).

**Comment by LeHenschel** [comment]
I had a similar issue. I think you can just change the greater to greater equal in this line https://github.com/Deep-MI/FastSurfer/blob/stable/FastSurferCNN/models/interpolation_layer.py#L250 to fix it.

`if out >= this_in_shape:` instead of if `out > this_in_shape:`

**Comment by ClePol** [comment]
Thank you, unfortunately that didn't fix it for me.

I believe I found out what caused my crash. If the scale factors are all one the normal procedure including padding is skipped: 
```
skip_interp = torch.all(torch.stack(scales, -1) == 1)`
```
It seems that due to some numerical inaccuracies it can happen that this is true during the first interpolation but not during the second. Then the error occurs. If I replace it with:
```
scales_comp = torch.stack(scales,-1)
skip_interp = torch.allclose(scales_comp, torch.ones(scales_comp.shape,dtype=scales_comp.dtype), atol=1e-08)
```
The error does not occur anymore.

**Comment by ClePol** [comment]
The skipping of cropping and padding overall is still a bug, I think - independent of the numerical issue.
